File: src/pages/agents_page.py
"""Agents page object (/lead-management/agents). Locators from LMS.pdf visible text."""

from playwright.sync_api import Page

from src.pages.base_page import BasePage


class AgentsPage(BasePage):
    def __init__(self, page: Page, base_url: str) -> None:
        super().__init__(page, base_url)
        # --- Elements (text/role sourced from LMS.pdf - no data-testid confirmed) ---
        self.team_avg_conversion_label = self.page.get_by_text("Team Average Conversion")
        self.team_avg_conversion_time_label = self.page.get_by_text(
            "Team Avg Conversion Time"
        )
        self.agents_above_average_label = self.page.get_by_text("Agents Above Average")
        self.agents_below_average_label = self.page.get_by_text("Agents Below Average")
        self.search_input = self.page.get_by_role("searchbox")
        self.filter_btn = self.page.get_by_role("button", name="Filter")
        self.agents_table = self.page.get_by_role("table")

    def goto_agents(self) -> None:
        self.goto("lead-management/agents")

File: src/pages/AgentsPage.ts
import type { Locator, Page } from '@playwright/test';

/** /lead-management/agents — "Agents". */
export class AgentsPage {
  readonly page: Page;

  readonly showFormerAgentsToggle: Locator;
  readonly teamAverageConversionValue: Locator;
  readonly teamAvgTimeOfConversionValue: Locator;

  static readonly EXPECTED_COLUMN_HEADERS = [
    'Agent Name',
    'Leads',
    'Converted',
    'Conv.Rate',
    'Overdue Leads',
    'Avg Conversion Time',
    'Performance',
  ] as const;

  constructor(page: Page) {
    this.page = page;
    this.showFormerAgentsToggle = page.getByText('Show former agents');
    this.teamAverageConversionValue = page.getByText('Team Average Conversion').locator('..');
    this.teamAvgTimeOfConversionValue = page.getByText('Team Avg Time of Conversion').locator('..');
  }

  async goto(): Promise<void> {
    await this.page.goto('/lead-management/agents', { waitUntil: 'domcontentloaded' });
  }

  async getColumnHeaderTexts(): Promise<string[]> {
    await this.page.getByText(AgentsPage.EXPECTED_COLUMN_HEADERS[0], { exact: true }).first().waitFor({ state: 'visible' });

    const texts: string[] = [];
    for (const header of AgentsPage.EXPECTED_COLUMN_HEADERS) {
      const locator = this.page.getByText(header, { exact: true }).first();
      // eslint-disable-next-line no-await-in-loop
      texts.push((await locator.isVisible()) ? header : '');
    }
    return texts;
  }
}
